velox/dwio/parquet/tests/writer/ParquetWriterTest.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/dwio/parquet/tests/ParquetTestBase.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::common;
+using namespace facebook::velox::dwio::common;
+using namespace facebook::velox::dwio::parquet;
+using namespace facebook::velox::parquet;
+
+class ParquetWriterTest : public ParquetTestBase {
+ protected:
+  std::unique_ptr<RowReader> createRowReaderWithSchema(
+      const std::unique_ptr<Reader> reader,
+      const RowTypePtr& rowType) {
+    auto rowReaderOpts = getReaderOpts(rowType);
+    auto scanSpec = makeScanSpec(rowType);
+    rowReaderOpts.setScanSpec(scanSpec);
+    auto rowReader = reader->createRowReader(rowReaderOpts);
+    return rowReader;
+  };
+};
+
+class SupportedParquetWriterCompressionTest
+    : public ParquetWriterTest,
+      public testing::WithParamInterface<CompressionKind> {
+ public:
+  SupportedParquetWriterCompressionTest() : compressionKind_(GetParam()) {}
+  static std::vector<CompressionKind> getTestParams() {
+    static std::vector<CompressionKind> params = {
+        CompressionKind::CompressionKind_NONE,
+        CompressionKind::CompressionKind_SNAPPY,
+    };
+    return params;
+  }
+
+ protected:
+  CompressionKind compressionKind_;
+};
+
+TEST_P(SupportedParquetWriterCompressionTest, compression) {
+  // Mock data
+  auto schema = ROW({"c0", "c1"}, {INTEGER(), DOUBLE()});
+
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(10, [](auto row) { return row + 5; }),
+      makeFlatVector<double>(10, [](auto row) { return row + 10; }),
+  });
+  auto filePath = fs::path(fmt::format("{}/test_close.txt", tempPath_->path));
+  auto [writer, sinkPtr] = createWriterWithSinkPtr(
+      filePath.string(),
+      [&]() {
+        return std::make_unique<LambdaFlushPolicy>(
+            kRowsInRowGroup, kBytesInRowGroup, [&]() { return false; });
+      },
+      compressionKind_);
+
+  writer->write(data);
+
+  writer->close();
+
+  dwio::common::ReaderOptions readerOptions{leafPool_.get()};
```

Comment:
If the data is compressed you would need to tell the reader also which compression it is. As it stands the test should fail because you are compressing the data in the writer.
Unless this passes but then it stands to reason that the data was not compressed.

Reply:
Makes sense. It does pass the tests. However, I am not really sure whether we can pass the  compression codec in the ReaderOptions.

Reply:
@czentgr, the compression used when writing is stored in the metadata of the file. So the reader does not have to be told what the compression is. It can infer that from the file metadata.